lib/config/config.go

```diff
@@ -375,6 +384,41 @@ func (cfg *Configuration) clean() error {
 	return nil
 }
 
+func convertV23V24(cfg *Configuration) {
+	var (
+		guiListener   = GUIListener{}
+		emptyListener = GUIListener{}
+	)
+	if cfg.GUI.Deprecated_RawAddress != "" {
+		guiListener.Address = cfg.GUI.Deprecated_RawAddress
+		cfg.GUI.Deprecated_RawAddress = ""
+	}
+	if cfg.GUI.Deprecated_RawUseTLS {
+		guiListener.UseTLS = true
+		cfg.GUI.Deprecated_RawUseTLS = false
+	}
+	if cfg.GUI.Deprecated_InsecureAdminAccess {
+		guiListener.InsecureAdminAccess = true
+		cfg.GUI.Deprecated_InsecureAdminAccess = false
+	}
+	if cfg.GUI.Deprecated_InsecureSkipHostCheck {
+		guiListener.InsecureSkipHostCheck = true
+		cfg.GUI.Deprecated_InsecureSkipHostCheck = false
+	}
+	if cfg.GUI.Deprecated_InsecureAllowFrameLoading {
+		guiListener.InsecureAllowFrameLoading = true
+		cfg.GUI.Deprecated_InsecureAllowFrameLoading = false
+	}
+
+	// Migrate if at least one deprecated field had been filled.  If both
+	// deprecated and new configuration style are used, merged them
+	// together.
+	if guiListener != emptyListener {
+		cfg.GUI.Listeners = append(cfg.GUI.Listeners, guiListener)
+	}
+	cfg.Version = 24
```

Comment:
So this is too complicated.
I'd just do
```
cfg.GUI.Listeners = append(cfg.GUI.Listeners, GUIListener{
    UseTLS: cfg.GUI.Deprecated_RawUseTLS
    ...
})
cfg.GUI.Deprecated_RawUseTLS = False
...
```

Reply:
You're so right ! ^^